server/src/routes/api/v2/contacts.js

```diff
@@ -0,0 +1,69 @@
+const router = require('express').Router()
+const mongoose = require('mongoose')
+const { ObjectId } = mongoose.Types
+const auth = require('@routes/auth')
+const UserWallet = mongoose.model('UserWallet')
+const Contact = mongoose.model('Contact')
+const { concat, compact, indexOf } = require('lodash')
+
+/**
+ * @api {post} /contacts/ Sync contacts list
+ * @apiName SyncContacts
+ * @apiGroup Contacts
+ * @apiDescription Sync contacts list
+ *
+ * @apiHeader {String} Authorization JWT Authorization in a format "Bearer {jwtToken}"
+ *
+ * @apiParam {String[]} contacts phone numbers list
+ *
+ * @apiSuccess {Object[]} new contacts list (phoneNumber, account)
+ * @apiSuccess {Number} nonce
+ */
+router.post('/', auth.required, async (req, res) => {
+  const { phoneNumber, accountAddress } = req.user
+  const { contacts } = req.body
+  const userWallet = await UserWallet.findOne({ phoneNumber, accountAddress }).populate('contacts')
+  const userWalletContactIds = userWallet.contacts.map(obj => obj._id)
+  const userWalletContactPhoneNumbers = userWallet.contacts.map(obj => obj.phoneNumber)
+  const nonce = (new Date()).getTime()
+
+  const newContactIds = compact(await Promise.all(contacts.map(phoneNumber => {
+    return new Promise(async (resolve, reject) => {
+      if (indexOf(userWalletContactPhoneNumbers, phoneNumber) >= 0) {
+        return resolve()
+      }
+      let contactUserWallet = await UserWallet.findOne({ phoneNumber: phoneNumber })
+      let contact = await new Contact({
+        userWallet: ObjectId(userWallet._id),
+        phoneNumber,
+        walletAddress: contactUserWallet && contactUserWallet.walletAddress,
+        state: contactUserWallet && contactUserWallet.walletAddress ? 'NEW' : 'EMPTY',
```

Comment:
i would user a cleaner syntax 
`walletAddress: contactUserWallet?.walletAddress,`
`state: contactUserWallet?.walletAddress? 'NEW' : 'EMPTY',`
I wouldn't comment on this if not the other comments

Reply:
It's a cleaner syntax but the linter that is set up on the studio doesn't like it :)